geomesa-z3/src/main/scala/org/locationtech/geomesa/conf/CommonProperties.scala

```diff
@@ -0,0 +1,74 @@
+/***********************************************************************
+ * Copyright (c) 2013-2019 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.conf
+
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.util.{Failure, Success, Try}
+
+
+/**
+  * <p></p>
+  *
+  * @author jianghongkang 2019年05月10日 20:11
+  * @version V1.0
+  */
+object CommonProperties extends LazyLogging {
+
+  val S2CoverConfig = SystemProperty("google.s2.coverer.config", "0,30,1,8")
+    .get.trim.split(",")
+    .map(item => item.toInt)
+  val S2MinLevel = S2CoverConfig(0)
```

Comment:
I don't think we should copy the SystemProperties code into the z3 module just to load these values. Instead, could we make them constructor properties on the S2 curve objects? Then these properties could be moved out to geomesa-index-api (or maybe there's a more appropriate place).

Reply:
this is actually also causing the compile to fail, as this module doesn't have the scala XML classes available.

Reply:
yeah, on second thought, this can actually be handled in a constructor properties.